python/ray/util/client/ray_client_helpers.py

```diff
@@ -38,6 +38,8 @@ def ray_start_client_server_pair(metadata=None, ray_connect_handler=None, **kwar
                 time.sleep(1)
                 if time.monotonic() - start > 30:
                     raise RuntimeError("Failed to terminate Ray")
+        # Allow windows to close processes before moving on
+        time.sleep(3)
```

Comment:
Nit: Should this be wrapped in an if statement so that it only happens on Windows?  Also, I wondered if this would slow things down excessively, but it looks like there are already some `sleep` statements above this line to account for Ray taking a few seconds to terminate.   Since this is just adding a few seconds more I think it's fine.